application-task-manager-default/src/main/java/com/xwiki/taskmanager/internal/TaskXDOMProcessor.java

```diff
@@ -154,90 +149,102 @@ public List<Task> extract(XDOM content, TaskCounter taskCounter)
      * @param taskObject the task object that will be used to update task macro.
      * @param context the current context.
      */
-    public void updateTask(DocumentReference documentReference, BaseObject taskObject, XWikiContext context)
+    public void updateTaskMacroCall(DocumentReference documentReference, BaseObject taskObject, XWikiContext context)
         throws XWikiException
     {
         XWikiDocument ownerDocument = context.getWiki().getDocument(documentReference, context).clone();
-        String taskId = taskObject.getStringValue(Task.ID);
+        if (ownerDocument.isNew()) {
+            return;
+        }
+        DocumentReference taskDocRef = taskObject.getDocumentReference();
         XDOM content = ownerDocument.getXDOM();
-        List<MacroBlock> macros = content.getBlocks(new MacroBlockMatcher(Task.NAME), Block.Axes.DESCENDANT);
+        List<MacroBlock> macros = content.getBlocks(new MacroBlockMatcher(Task.MACRO_NAME), Block.Axes.DESCENDANT);
         SimpleDateFormat storageFormat = new SimpleDateFormat(configuration.getStorageDateFormat());
         for (MacroBlock macro : macros) {
-            if (macro.getParameters().getOrDefault(Task.ID, "").equals(taskId)) {
+            DocumentReference macroRef = resolver.resolve(macro.getParameters().getOrDefault(Task.REFERENCE, ""));
```

Comment:
```suggestion
            DocumentReference macroRef = resolver.resolve(macro.getParameters().getOrDefault(Task.REFERENCE, ""), documentReference);
```

The task reference (from the macro call) should be resolved relative to the updated document (that holds the macro call).